gobblin-api/src/test/java/gobblin/testing/TestingUtilsTest.java

```diff
@@ -0,0 +1,108 @@
+  }
+
+  @Test
+  public void testAssertWithBackoff_conditionTrue() throws Exception {
+    Logger log = LoggerFactory.getLogger("testAssertWithBackoff_conditionTrue");
+    TestingUtils.assertWithBackoff(new Callable<Boolean>() {
+        @Override public Boolean call() { return true; }
+      }, 1000L, "should always succeed", log);
+  }
+
+  @Test
+  public void testAssertWithBackoff_conditionEventuallyTrue() throws Exception {
+    Logger log = LoggerFactory.getLogger("testAssertWithBackoff_conditionEventuallyTrue");
+    //TestingUtils.setLogjLevelForLogger(log, Level.DEBUG);
+    final AtomicInteger cnt = new AtomicInteger();
+    TestingUtils.assertWithBackoff(new Callable<Boolean>() {
+        @Override public Boolean call() { return cnt.incrementAndGet() >= 6; }
+      }, 1000L, Optional.of("should eventually succeed"),
+      Optional.of(log), Optional.of(2.0), Optional.of(100L));
+  }
+
+  @Test
+  public void testAssertWithBackoff_conditionFalse() throws Exception {
+    Logger log = LoggerFactory.getLogger("testAssertWithBackoff_conditionFalse");
+    //setLogjLevelForLogger(log, Level.DEBUG);
+    long startTimeMs = System.currentTimeMillis();
+    try {
+      TestingUtils.assertWithBackoff(new Callable<Boolean>() {
+          @Override public Boolean call() { return false; }
+        }, 50L, Optional.of("should timeout"),
+        Optional.of(log), Optional.<Double>absent(), Optional.<Long>absent());
+      Assert.fail("TimeoutException expected");
+    } catch (TimeoutException e) {
+      //Expected
+    }
+    long durationMs = System.currentTimeMillis() - startTimeMs;
+    log.debug("assert took " + durationMs + "ms");
+    Assert.assertTrue(durationMs >= 50L, Long.toString(durationMs) + ">= 50ms");
+  }
+
+  @Test
+  public void testAssertWithBackoff_InterruptedException() throws Exception {
+    try {
+      TestingUtils.assertWithBackoff(new Callable<Boolean>() {
+          @Override public Boolean call() throws Exception { throw new InterruptedException(); }
+        }, 50L, "should throw InterruptedException");
+      Assert.fail("InterruptedException expected");
+    } catch (InterruptedException e) {
+      //Expected
+    }
+  }
+
+  @Test
+  public void testAssertWithBackoff_RuntimeException() throws Exception {
+    Logger log = LoggerFactory.getLogger("testAssertWithBackoff_RuntimeException");
+    //setLogjLevelForLogger(log, Level.DEBUG);
+    try {
+      TestingUtils.assertWithBackoff(new Callable<Boolean>() {
+          @Override public Boolean call() throws Exception { throw new Exception("BLAH"); }
+        }, 50L, Optional.of("should throw RuntimeException"),
+        Optional.of(log), Optional.<Double>absent(), Optional.<Long>absent());
+      Assert.fail("should throw RuntimeException");
+    } catch (RuntimeException e) {
+      Assert.assertTrue(e.getMessage().indexOf("BLAH") > 0, e.getMessage());
+    }
+  }
+
+  public static void setLogjLevelForLogger(Logger log, Level logLevel) {
```

Comment:
Why does the log level need to be changed? It is for testing and not so important, but this is asking for trouble. If we ever decide to change logging framework, this would silently stop working.